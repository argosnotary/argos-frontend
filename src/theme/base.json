{
  "fontFamily": "'Fira Sans'",
  "fontFile": "/fonts/fira-sans-v10-latin-regular",
  "button": {
    "bgColor": "#1779ba",
    "textColor": "#fff"
  },
  "cancelButton": {
    "bgColor": "transparent",
    "textColor": "#8D99AE",
    "borderColor": "#8D99AE",
    "hover": {
      "bgColor": "#8D99AE",
      "textColor": "#fff",
      "borderColor": "transparent"
    }
  },
  "transparentButton": {
    "default": {
      "transparentButtonBgColor": "#fff",
      "transparentButtonTextColor": "#1779ba",
      "transparentButtonBorderColor": "#1779ba"
    },
    "hover": {
      "transparentButtonBgColor": "#1779ba",
      "transparentButtonTextColor": "#fff",
      "transparentButtonBorderColor": "#1779ba"
    }
  },
  "alternateLoader": {
    "color": "#808080"
  },
  "loaderIcon": {
    "color": "#1779ba"
  },
  "loaderButton": {
    "loadingColor": "#fff"
  },
  "dataCheckbox": {
    "loaderColor": "#1779ba"
  },
  "searchInput": {
    "searchResultsBgColor": "#fff",
    "searchResultEntryHoverBg": "#e8feff",
    "selectedEntryBorderColor": "#e0e0e0",
    "searchResultsBorderColor": "#e0e0e0",
    "loaderIconColor": "#1779ba"
  },
  "collapsibleContainer": {
    "iconColor": "gray",
    "headerBgColor": "#e0e0e0",
    "bodyBorderColor": "#e0e0e0"
  },
  "selectList": {
    "defaultItemBgColor": "#fff",
    "selectedItemBgColor": "#e8feff",
    "iconColor": "#8d99ae"
  },
  "panel": {
    "icons": {
      "shrinkIcon": {
        "color": "#1779ba"
      },
      "enlargeIcon": {
        "color": "#1779ba"
      }
    }
  },
  "modal": {
    "bgColor": "#fff",
    "button": {
      "default": {
        "color": "#1779ba"
      },
      "hover": {
        "color": "#1779ba"
      }
    }
  },
  "passwordView": {
    "clipboardIcon": {
      "color": "#8D99AE",
      "hoverBgColor": "#e8feff"
    }
  },
  "copyInput": {
    "bgColor": "#e0e0e0"
  },
  "notificationsList": {
    "notificationBgColor": "#fff",
    "iconSize": 22,
    "iconColor": "#fff",
    "colors": {
      "WARNING": "orange",
      "ERROR": "red",
      "SUCCESS": "green"
    }
  },
  "layoutBuilder": {
    "segmentContainerBorderColor": "#1779ba",
    "segmentContainerTitleBgColor": "#1779ba",
    "iconColor": "#1779ba",
    "stepContainerBgColor": "#e0e0e0",
    "stepContainerTitleBorderColor": "#1779ba",
    "addStepButtonBorderColor": "#1779ba",
    "segmentTitleBgColor": "#fff",
    "segmentTitleHoverBorderColor": "#1779ba",
    "stepTitleBgColor": "#fff",
    "stepTitleHoverBorderColor": "#1779ba",
    "segmentsContainerTitleColor": "#fff"
  },
  "treeEditor": {
    "textColor": "#2e2e2e",
    "bgColor": "#f1f1f1",
    "borderColor": "#e0e0e0",
    "loaders": {
      "onPageLoad": {
        "color": "#1779ba"
      },
      "onFetchChildren": {
        "color": "#1779ba"
      }
    },
    "iconColors": {
      "addRootNode": "#1779ba",
      "expandNode": "#1779ba",
      "label": "#8D99AE",
      "chain": "#8D99AE",
      "robot": "#8D99AE"
    },
    "treeHeadLabel": {
      "bgColor": "#e8feff"
    },
    "nodeContextMenuContainer": {
      "bgColor": "fff",
      "borderColor": "#e5e5e5"
    },
    "nodeContextMenuItem": {
      "iconColor": "#8D99AE",
      "bgColor": "#fff",
      "hover": {
        "bgColor": "#e8feff"
      }
    },
    "breadCrumb": {
      "textColor": "#1779ba"
    },
    "lastBreadCrumb": {
      "textColor": "#000"
    },
    "nodeContextMenuItemSeparator": {
      "borderColor": "#e5e5e5"
    },
    "nodeContextMenuIconSeparator": {
      "borderColor": "#8D99AE"
    }
  },
  "dashboardNavBar": {
    "link": {
      "textColor": "#2e2e2e",
      "hover": {
        "bgColor": "#e8feff"
      }
    }
  },
  "connectionError": {
    "textColor": "#fff",
    "bgColor": "red"
  },
  "alerts": {
    "warning": {
      "color": "red"
    }
  },
  "PageHeader": {
    "bottomBorderColor": "#bbb"
  },
  "homePage": {
    "bgColor": "#eee",
    "navigation": {
      "bgColor": "#fff",
      "loginButton": {
        "default": {
          "loginButtonBgColor": "#fff",
          "loginButtonTextColor": "#1779ba",
          "loginButtonBorderColor": "#1779ba"
        },
        "hover": {
          "loginButtonBgColor": "#1779ba",
          "loginButtonTextColor": "#fff",
          "loginButtonBorderColor": "#1779ba"
        }
      }
    }
  },
  "dashboardPage": {
    "avatar": {
      "borderColor": "#1779ba"
    },
    "contextMenu": {
      "bgColor": "#fff",
      "borderColor": "#e5e5e5",
      "menuItemTextColor": "#2e2e2e",
      "menuItemBgColor": "#fff",
      "menuItemHoverBgColor": "#eee",
      "dividerBgColor": "#dfdfdf"
    }
  },
  "loginPage": {
    "bgColor": "#eee"
  },
  "userSettingsPage": {
    "bgColor": "#f1f1f1",
    "sidePanel": {
      "bgColor": "#e0e0e0",
      "sidePanelItem": {
        "bgColor": "#c7c7c7",
        "activeBgColor": "#f1f1f1",
        "highlightColor": "#1779ba",
        "fontColor": "#2E2E30"
      }
    }
  },
  "keyManagementPage": {
    "iconColor": "#1779ba",
    "passwordColor": "orange",
    "noActiveKeyWarning": {
      "textColor": "orange",
      "borderColor": "orange"
    }
  },
  "layoutPage": {
    "panel": {
      "bgColor": "#f1f1f1",
      "borderColor": "#e0e0e0",
      "contentSeparator": "#ccc"
    }
  }
}
