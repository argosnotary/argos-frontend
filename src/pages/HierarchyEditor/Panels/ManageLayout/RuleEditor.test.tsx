/*
 * Copyright (C) 2020 Argos Notary CoÃ¶peratie UA
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React from "react";
import { mount, ReactWrapper } from "enzyme";

import { act } from "react-dom/test-utils";
import { waitFor } from "@testing-library/dom";
import RuleEditor from "./RuleEditor";
import {
  DetailsPanelType,
  ILayoutEditorStoreContext,
  LayoutEditorActionType,
  LayoutEditorStoreContext
} from "../../../../stores/LayoutEditorStore";
import {
  ILayout,
  IRule,
  IStep,
  RuleDestinationTypeEnum,
  RuleRuleTypeEnum
} from "../../../../interfaces/ILayout";
import theme from "../../../../theme/base.json";
import { ThemeProvider } from "styled-components";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useHistory: () => ({
    push: jest.fn()
  })
}));

const dispatch = jest.fn();

const editorStoreContextExpectedEndProduct: ILayoutEditorStoreContext = {
  state: {
    layout: {
      expectedEndProducts: [
        {
          pattern: "pattern",
          destinationSegmentName: "seg1",
          sourcePathPrefix: "sourcePathPrefix",
          destinationStepName: "step1",
          destinationType: RuleDestinationTypeEnum.MATERIALS,
          destinationPathPrefix: "destinationPathPrefix",
          ruleType: RuleRuleTypeEnum.MATCH
        }
      ],
      layoutSegments: [
        {
          name: "seg1",
          steps: [
            {
              name: "step1"
            },
            {
              name: "step2"
            }
          ]
        },
        {
          name: "seg2",
          steps: [
            {
              name: "stepa"
            },
            {
              name: "stepb"
            }
          ]
        }
      ]
    } as ILayout,
    detailPanelMode: DetailsPanelType.LAYOUT_DETAILS,
    approvalConfigs: [],
    showSigningDialog: false,
    loading: false,
    activeEditLayoutElement: undefined,
    selectedLayoutElement: undefined
  },
  dispatch: dispatch
};

const editorStoreContextStepRule: ILayoutEditorStoreContext = {
  state: {
    layout: {
      layoutSegments: [
        {
          name: "seg1",
          steps: [
            {
              name: "step1"
            },
            {
              name: "step2"
            }
          ]
        },
        {
          name: "seg2",
          steps: [
            {
              name: "step3"
            },
            {
              name: "step4"
            }
          ]
        }
      ]
    } as ILayout,
    detailPanelMode: DetailsPanelType.LAYOUT_DETAILS,
    approvalConfigs: [],
    showSigningDialog: false,
    loading: false,
    activeEditLayoutElement: undefined,
    selectedLayoutElement: {
      segment: {
        name: "seg1",
        steps: [
          {
            name: "step1"
          },
          {
            name: "step2"
          } as IStep
        ]
      },
      step: {
        name: "step1",
        expectedMaterials: [
          {
            ruleType: RuleRuleTypeEnum.CREATE,
            pattern: "pattern1"
          }
        ]
      } as IStep
    }
  },
  dispatch: dispatch
} as ILayoutEditorStoreContext;

function createComponent(
  editorStoreContext: ILayoutEditorStoreContext,
  initialRules?: Array<IRule>
) {
  return mount(
    <ThemeProvider theme={theme}>
      <LayoutEditorStoreContext.Provider value={editorStoreContext}>
        <RuleEditor
          title={"Expected End Products"}
          initialRules={initialRules}
          addAction={LayoutEditorActionType.ADD_EXPECTED_END_PRODUCT}
          editAction={LayoutEditorActionType.EDIT_EXPECTED_END_PRODUCT}
          removeAction={LayoutEditorActionType.REMOVE_EXPECTED_END_PRODUCT}
        />
      </LayoutEditorStoreContext.Provider>
    </ThemeProvider>
  );
}

it("renders correctly", async () => {
  const root = createComponent(
    editorStoreContextExpectedEndProduct,
    editorStoreContextExpectedEndProduct.state.layout.expectedEndProducts
  );

  await act(async () => {
    await waitFor(() => {
      root.update();
      return expect(
        root.find('button[data-testhook-id="edit-rule-0"]').length
      ).toBe(1);
    });

    expect(root.find(RuleEditor)).toMatchSnapshot();
  });
});

it("add rule expected end product", async () => {
  dispatch.mockReset();
  const root = createComponent(
    editorStoreContextExpectedEndProduct,
    editorStoreContextExpectedEndProduct.state.layout.expectedEndProducts
  );

  await act(async () => {
    await waitFor(() => {
      root.update();
      return expect(
        root.find('button[data-testhook-id="edit-rule-0"]').length
      ).toBe(1);
    });

    root.find('button[data-testhook-id="add-rule"]').simulate("click");

    root.find('form[data-testhook-id="rule-edit-form"]').simulate("submit");

    expect(root.find(RuleEditor)).toMatchSnapshot();

    updateField(
      root.find('input[data-testhook-id="rule-edit-form-field-0"]'),
      "pattern",
      "pattern1"
    );

    updateField(
      root.find('input[data-testhook-id="rule-edit-form-field-1"]'),
      "sourcePathPrefix",
      "sourcePathPrefix1"
    );

    updateField(
      root.find('input[data-testhook-id="rule-edit-form-field-2"]'),
      "destinationPathPrefix",
      "destinationPathPrefix1"
    );

    updateField(
      root.find('select[data-testhook-id="rule-edit-form-field-3"]'),
      "destinationType",
      "PRODUCTS"
    );

    updateField(
      root.find('select[data-testhook-id="rule-edit-form-field-4"]'),
      "destinationSegmentName",
      "seg2"
    );

    updateField(
      root.find('select[data-testhook-id="rule-edit-form-field-5"]'),
      "destinationStepName",
      "step4"
    );

    root.find('form[data-testhook-id="rule-edit-form"]').simulate("submit");

    await waitFor(() => {
      root.update();
      return expect(dispatch.mock.calls.length).toBe(1);
    });

    expect(dispatch.mock.calls[0][0]).toEqual({
      rule: {
        destinationPathPrefix: "destinationPathPrefix1",
        destinationSegmentName: "seg2",
        destinationStepName: "step4",
        destinationType: "PRODUCTS",
        pattern: "pattern1",
        sourcePathPrefix: "sourcePathPrefix1"
      },
      type: 21
    });
  });
});

it("add rule allow", async () => {
  dispatch.mockReset();
  const root = createComponent(
    editorStoreContextStepRule,
    editorStoreContextStepRule.state.selectedLayoutElement?.step
      ?.expectedMaterials
  );

  await act(async () => {
    await waitFor(() => {
      root.update();
      return expect(
        root.find('button[data-testhook-id="edit-rule-0"]').length
      ).toBe(1);
    });

    root.find('button[data-testhook-id="add-rule"]').simulate("click");

    await waitFor(() => {
      root.find('select[id="ruleType"]').simulate("change", {
        target: {
          name: "ruleType",
          value: "ALLOW"
        }
      });
      root.update();

      expect(root.find('select[id="ruleType"]').props().value).toBe("ALLOW");
    });

    root.find('form[data-testhook-id="rule-edit-form"]').simulate("submit");

    expect(root.find(RuleEditor)).toMatchSnapshot();

    updateField(
      root.find('input[data-testhook-id="rule-edit-form-field-0"]'),
      "pattern",
      "pattern2"
    );

    root.find('form[data-testhook-id="rule-edit-form"]').simulate("submit");

    await waitFor(() => {
      root.update();
      return expect(dispatch.mock.calls.length).toBe(1);
    });

    expect(dispatch.mock.calls[0][0]).toEqual({
      rule: {
        pattern: "pattern2",
        ruleType: "ALLOW"
      },
      type: 21
    });
  });
});

const updateField = (wrapper: ReactWrapper<any>, name: string, value: any) => {
  wrapper.simulate("input", {
    persist: () => {},
    target: {
      name,
      value
    }
  });
  wrapper.simulate("change", {
    persist: () => {},
    target: {
      name,
      value
    }
  });
};
