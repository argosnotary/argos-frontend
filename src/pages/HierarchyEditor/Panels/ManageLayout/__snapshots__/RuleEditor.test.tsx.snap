// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RuleEditor>
  <Styled(styled.section)>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-jKJlTe",
            "isStatic": false,
            "lastClassName": "gtWwp",
            "rules": Array [
              "
  margin: 1rem 0 1rem;
  border: 1px solid #1779ba;
  display: flex;
  align-items: center;
  width: 100%;
  min-height: 18.8rem;
",
              "
  min-height: 0;
  flex-direction: column;
  border: 0;
  padding: 0 1rem 1rem;
  border: 1px solid
    ",
              [Function],
              ";
",
            ],
          },
          "displayName": "Styled(styled.section)",
          "foldedComponentIds": Array [
            "sc-bdVaJa",
          ],
          "render": [Function],
          "styledComponentId": "sc-jKJlTe",
          "target": "section",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <section
        className="sc-bdVaJa sc-jKJlTe gtWwp"
      >
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-ifAKCX",
                  "isStatic": false,
                  "lastClassName": "cBkhpN",
                  "rules": Array [
                    "
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  width: 100%;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-ifAKCX",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-ifAKCX cBkhpN"
            >
              <Styled(styled.header)>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-eNQAEJ",
                        "isStatic": false,
                        "lastClassName": "jxoviP",
                        "rules": Array [
                          "
  position: relative;
  font-size: 0.9rem;
  top: -0.75rem;
  background-color: #f1f1f1;
  display: inline-flex;
",
                          "
  font-size: 0.85rem;
  top: -1rem;
  color: ",
                          [Function],
                          ";
  background-color: ",
                          [Function],
                          ";
  padding: 0.25rem 2rem 0.4rem;
",
                        ],
                      },
                      "displayName": "Styled(styled.header)",
                      "foldedComponentIds": Array [
                        "sc-EHOje",
                      ],
                      "render": [Function],
                      "styledComponentId": "sc-eNQAEJ",
                      "target": "header",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <header
                    className="sc-EHOje sc-eNQAEJ jxoviP"
                  >
                    Expected End Products
                  </header>
                </StyledComponent>
              </Styled(styled.header)>
              <Styled(styled.button)
                data-testhook-id="add-rule"
                onClick={[Function]}
              >
                <StyledComponent
                  data-testhook-id="add-rule"
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-hMqMXs",
                        "isStatic": false,
                        "lastClassName": "PCSBs",
                        "rules": Array [
                          "
  position: relative;
  top: -1rem;
  background-color: #f1f1f1;
  border: 0;
",
                          "
  right: 0;

  &:hover {
    cursor: pointer;
    transform: scale(0.8);
  }
",
                        ],
                      },
                      "displayName": "Styled(styled.button)",
                      "foldedComponentIds": Array [
                        "sc-bxivhb",
                      ],
                      "render": [Function],
                      "styledComponentId": "sc-hMqMXs",
                      "target": "button",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  onClick={[Function]}
                >
                  <button
                    className="sc-bxivhb sc-hMqMXs PCSBs"
                    data-testhook-id="add-rule"
                    onClick={[Function]}
                  >
                    <PlusIcon
                      color="#1779ba"
                      size={24}
                    >
                      <svg
                        height={24}
                        version="1.1"
                        viewBox="0 0 32 32"
                        width={24}
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <title>
                          plus
                        </title>
                        <path
                          d="M31 12h-11v-11c0-0.552-0.448-1-1-1h-6c-0.552 0-1 0.448-1 1v11h-11c-0.552 0-1 0.448-1 1v6c0 0.552 0.448 1 1 1h11v11c0 0.552 0.448 1 1 1h6c0.552 0 1-0.448 1-1v-11h11c0.552 0 1-0.448 1-1v-6c0-0.552-0.448-1-1-1z"
                          fill="#1779ba"
                        />
                      </svg>
                    </PlusIcon>
                  </button>
                </StyledComponent>
              </Styled(styled.button)>
            </div>
          </StyledComponent>
        </styled.div>
        <styled.ul>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bZQynM",
                  "isStatic": false,
                  "lastClassName": "enhIWo",
                  "rules": Array [
                    "
  display: flex;
  flex-direction: column;
  width: 100%;
",
                  ],
                },
                "displayName": "styled.ul",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bZQynM",
                "target": "ul",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <ul
              className="sc-bZQynM enhIWo"
            >
              <li
                key="rule-row-0"
              >
                <styled.section>
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-kkGfuU",
                          "isStatic": false,
                          "lastClassName": "cwDmEf",
                          "rules": Array [
                            "
  width: 100%;
  margin: 0 0 1rem;
",
                          ],
                        },
                        "displayName": "styled.section",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-kkGfuU",
                        "target": "section",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <section
                      className="sc-kkGfuU cwDmEf"
                    >
                      <styled.header>
                        <StyledComponent
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "sc-kEYyzF",
                                "isStatic": false,
                                "lastClassName": "jPwoGc",
                                "rules": Array [
                                  "
  border: 1px solid transparent;
  box-sizing: border-box;
  padding: 0.5rem;
  width: 100%;
  margin: 0.2rem 0 0;
  background-color: ",
                                  [Function],
                                  ";

  display: flex;
  align-items: center;
  justify-content: space-between;

  span {
    margin: 0 0.5rem;
  }
",
                                ],
                              },
                              "displayName": "styled.header",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "sc-kEYyzF",
                              "target": "header",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                        >
                          <header
                            className="sc-kEYyzF jPwoGc"
                          >
                            <styled.span>
                              <StyledComponent
                                forwardedComponent={
                                  Object {
                                    "$$typeof": Symbol(react.forward_ref),
                                    "attrs": Array [],
                                    "componentStyle": ComponentStyle {
                                      "componentId": "sc-gzVnrw",
                                      "isStatic": false,
                                      "lastClassName": "bCKUNd",
                                      "rules": Array [
                                        "
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
",
                                      ],
                                    },
                                    "displayName": "styled.span",
                                    "foldedComponentIds": Array [],
                                    "render": [Function],
                                    "styledComponentId": "sc-gzVnrw",
                                    "target": "span",
                                    "toString": [Function],
                                    "warnTooManyClasses": [Function],
                                    "withComponent": [Function],
                                  }
                                }
                                forwardedRef={null}
                              >
                                <span
                                  className="sc-gzVnrw bCKUNd"
                                >
                                  match - pattern - destinationPathPrefix
                                </span>
                              </StyledComponent>
                            </styled.span>
                            <styled.div>
                              <StyledComponent
                                forwardedComponent={
                                  Object {
                                    "$$typeof": Symbol(react.forward_ref),
                                    "attrs": Array [],
                                    "componentStyle": ComponentStyle {
                                      "componentId": "sc-bwzfXH",
                                      "isStatic": false,
                                      "lastClassName": "fNOhzS",
                                      "rules": Array [
                                        "
  display: flex;
  align-items: center;
",
                                      ],
                                    },
                                    "displayName": "styled.div",
                                    "foldedComponentIds": Array [],
                                    "render": [Function],
                                    "styledComponentId": "sc-bwzfXH",
                                    "target": "div",
                                    "toString": [Function],
                                    "warnTooManyClasses": [Function],
                                    "withComponent": [Function],
                                  }
                                }
                                forwardedRef={null}
                              >
                                <div
                                  className="sc-bwzfXH fNOhzS"
                                >
                                  <Styled(styled.button)
                                    data-testhook-id="edit-rule-0"
                                    onClick={[Function]}
                                  >
                                    <StyledComponent
                                      data-testhook-id="edit-rule-0"
                                      forwardedComponent={
                                        Object {
                                          "$$typeof": Symbol(react.forward_ref),
                                          "attrs": Array [],
                                          "componentStyle": ComponentStyle {
                                            "componentId": "sc-hSdWYo",
                                            "isStatic": false,
                                            "lastClassName": "epMMcD",
                                            "rules": Array [
                                              "
  display: inline-flex;
  background: none;
  border: 0;

  &:hover {
    cursor: pointer;
    transform: scale(0.8);
  }
",
                                            ],
                                          },
                                          "displayName": "Styled(styled.button)",
                                          "foldedComponentIds": Array [
                                            "sc-htpNat",
                                          ],
                                          "render": [Function],
                                          "styledComponentId": "sc-hSdWYo",
                                          "target": "button",
                                          "toString": [Function],
                                          "warnTooManyClasses": [Function],
                                          "withComponent": [Function],
                                        }
                                      }
                                      forwardedRef={null}
                                      onClick={[Function]}
                                    >
                                      <button
                                        className="sc-htpNat sc-hSdWYo epMMcD"
                                        data-testhook-id="edit-rule-0"
                                        onClick={[Function]}
                                      >
                                        <EditIcon
                                          color="#1779ba"
                                          size={26}
                                        >
                                          <svg
                                            height={26}
                                            version="1.1"
                                            viewBox="0 0 24 24"
                                            width={26}
                                            xmlns="http://www.w3.org/2000/svg"
                                          >
                                            <path
                                              d="M21.561 5.318l-2.879-2.879c-0.293-0.293-0.677-0.439-1.061-0.439-0.385 0-0.768 0.146-1.061 0.439l-3.56 3.561h-9c-0.552 0-1 0.447-1 1v13c0 0.553 0.448 1 1 1h13c0.552 0 1-0.447 1-1v-9l3.561-3.561c0.293-0.293 0.439-0.677 0.439-1.061s-0.146-0.767-0.439-1.060zM11.5 14.672l-2.172-2.172 6.293-6.293 2.172 2.172-6.293 6.293zM8.939 13.333l1.756 1.728-1.695-0.061-0.061-1.667zM16 19h-11v-11h6l-3.18 3.18c-0.293 0.293-0.478 0.812-0.629 1.289-0.16 0.5-0.191 1.056-0.191 1.47v3.061h3.061c0.414 0 1.108-0.1 1.571-0.29 0.464-0.19 0.896-0.347 1.188-0.64l3.18-3.070v6zM18.5 7.672l-2.172-2.172 1.293-1.293 2.171 2.172-1.292 1.293z"
                                              fill="#1779ba"
                                            />
                                          </svg>
                                        </EditIcon>
                                      </button>
                                    </StyledComponent>
                                  </Styled(styled.button)>
                                  <Styled(styled.button)
                                    data-testhook-id="delete-rule-0"
                                    onClick={[Function]}
                                  >
                                    <StyledComponent
                                      data-testhook-id="delete-rule-0"
                                      forwardedComponent={
                                        Object {
                                          "$$typeof": Symbol(react.forward_ref),
                                          "attrs": Array [],
                                          "componentStyle": ComponentStyle {
                                            "componentId": "sc-iAyFgw",
                                            "isStatic": false,
                                            "lastClassName": "gwNQkp",
                                            "rules": Array [
                                              "
  display: inline-flex;
  background: none;
  border: 0;

  &:hover {
    cursor: pointer;
    transform: scale(0.8);
  }
",
                                            ],
                                          },
                                          "displayName": "Styled(styled.button)",
                                          "foldedComponentIds": Array [
                                            "sc-htpNat",
                                          ],
                                          "render": [Function],
                                          "styledComponentId": "sc-iAyFgw",
                                          "target": "button",
                                          "toString": [Function],
                                          "warnTooManyClasses": [Function],
                                          "withComponent": [Function],
                                        }
                                      }
                                      forwardedRef={null}
                                      onClick={[Function]}
                                    >
                                      <button
                                        className="sc-htpNat sc-iAyFgw gwNQkp"
                                        data-testhook-id="delete-rule-0"
                                        onClick={[Function]}
                                      >
                                        <RemoveIcon
                                          color="#1779ba"
                                          size={24}
                                        >
                                          <svg
                                            height={24}
                                            version="1.1"
                                            viewBox="0 0 32 32"
                                            width={24}
                                            xmlns="http://www.w3.org/2000/svg"
                                          >
                                            <title>
                                              trash
                                            </title>
                                            <path
                                              d="M29.98 6.819c-0.096-1.57-1.387-2.816-2.98-2.816h-3v-1.002c0-1.657-1.344-3-3-3h-10c-1.657 0-3 1.343-3 3v1.001h-3c-1.595 0-2.885 1.246-2.981 2.816h-0.019v2.183c0 1.104 0.896 2 2 2v0 17c0 2.209 1.791 4 4 4h16c2.209 0 4-1.791 4-4v-17c1.104 0 2-0.896 2-2v-2.182h-0.020zM10 3.002c0-0.553 0.447-1 1-1h10c0.553 0 1 0.447 1 1v1h-12v-1zM26 28.002c0 1.102-0.898 2-2 2h-16c-1.103 0-2-0.898-2-2v-17h20v17zM28 8.001v1h-24v-1.999c0-0.553 0.447-1 1-1h22c0.553 0 1 0.447 1 1v0.999zM9 28.006h2c0.553 0 1-0.447 1-1v-13c0-0.553-0.447-1-1-1h-2c-0.553 0-1 0.447-1 1v13c0 0.553 0.447 1 1 1zM9 14.005h2v13h-2v-13zM15 28.006h2c0.553 0 1-0.447 1-1v-13c0-0.553-0.447-1-1-1h-2c-0.553 0-1 0.447-1 1v13c0 0.553 0.447 1 1 1zM15 14.005h2v13h-2v-13zM21 28.006h2c0.553 0 1-0.447 1-1v-13c0-0.553-0.447-1-1-1h-2c-0.553 0-1 0.447-1 1v13c0 0.553 0.447 1 1 1zM21 14.005h2v13h-2v-13z"
                                              fill="#1779ba"
                                            />
                                          </svg>
                                        </RemoveIcon>
                                      </button>
                                    </StyledComponent>
                                  </Styled(styled.button)>
                                </div>
                              </StyledComponent>
                            </styled.div>
                          </header>
                        </StyledComponent>
                      </styled.header>
                    </section>
                  </StyledComponent>
                </styled.section>
              </li>
            </ul>
          </StyledComponent>
        </styled.ul>
      </section>
    </StyledComponent>
  </Styled(styled.section)>
</RuleEditor>
`;
